testnet/src/neon_node.rs

```diff
@@ -515,6 +557,27 @@ impl InitializedNeonNode {
         true
     }
 
+    fn relayer_mint_microblocks(mined_block_bhh: &BurnchainHeaderHash,
+                                mined_block_shh: &BlockHeaderHash,
+                                chain_state: &mut StacksChainState,
+                                keychain: &Keychain,
+                                consumed_execution: ExecutionCost,
+                                mem_pool: &MemPoolDB) -> Result<StacksMicroblock, ChainstateError> {
+        let mut microblock_miner = StacksMicroblockBuilder::new(mined_block_shh.clone(),
+                                                                mined_block_bhh.clone(),
+                                                                chain_state,
+                                                                consumed_execution)?;
+        let mblock_key = keychain.get_microblock_key()
```

Comment:
Each microblock public key hash must be unique in the chain history.  Not sure this is happening here.

Reply:
Yep -- the microblock pubkey hash is rotated before building the anchor block (if it rotated during the microblock assembly, it wouldn't match the the key from the anchor block).